src/components/TaskModal/TaskModal.tsx

```diff
@@ -0,0 +1,220 @@
+import { Box, Button, Grid, MenuItem, Modal, TextField } from '@mui/material';
+import { useFormik } from 'formik';
+import { Task, TaskPriority } from '@interfaces/task';
+import { TEST_USER } from '@interfaces/user';
+
+export type ModalMode = 'Create' | 'Edit';
+
+interface TaskModal {
+    modalMode: ModalMode;
+    isModalOpen: boolean;
+    handleModalClose: () => void;
+    createNewTask: (task: Task) => void;
+    editTask: (task: Task) => void;
+    existingTask?: Task;
+}
+
+function TaskModal({
+    modalMode,
+    isModalOpen,
+    handleModalClose,
+    createNewTask,
+    editTask,
+    existingTask,
+}: TaskModal) {
+    const style = {
+        position: 'absolute' as 'absolute',
+        top: '50%',
+        left: '50%',
+        transform: 'translate(-50%, -50%)',
+        width: 400,
+        bgcolor: 'background.paper',
+        border: '2px solid #000',
+        boxShadow: 24,
+        p: 4,
+    };
+
+    const {
+        handleChange,
+        handleSubmit,
+        resetForm,
+        values: { title, description, assignee, project, priority, deadline },
+    } = useFormik<{
+        title: string;
+        description: string;
+        assignee: string;
+        priority: TaskPriority;
+        deadline: Date | null;
+        project: string;
+    }>({
+        initialValues:
+            modalMode === 'Edit' && existingTask
+                ? {
+                      title: existingTask!.title,
+                      description: existingTask!.description,
+                      assignee: existingTask!.assignee,
+                      priority: existingTask!.priority,
+                      deadline: existingTask!.deadline,
+                      project: existingTask!.project,
+                  }
+                : {
+                      title: '',
+                      description: '',
+                      assignee: '',
+                      priority: 'Low',
+                      deadline: null,
+                      project: '',
+                  },
```

Comment:
you can save some space here by going `title: existingTask?.title || ''` the optional chaining will return false if existingTask is undefined and set title to ''